src/03-ticket-calculator.js

```diff
@@ -109,7 +134,84 @@ function calculateTicketPrice(ticketData, ticketInfo) {}
     purchaseTickets(tickets, purchases);
     //> "Ticket type 'discount' cannot be found."
  */
-function purchaseTickets(ticketData, purchases) {}
+function purchaseTickets(ticketData, purchases) {
+
+  // A string that will be used to return the full receipt
+  let receiptDetails = "";
+  let totalPrice = 0;
+
+  // Loop through the purchases array to access each object element
+  for (let i = 0; i < purchases.length; i++) {
+    // Variable for holding the price for each ticket
+    let ticketPrice = 0;
+    // String that will contain the extras in the format of (Terrace Access, Education Access)
+    let extraString = "";
+
+    // Return an error message if the ticketType does not exist as an object name inside of ticketData
+    if (!ticketData[purchases[i].ticketType]) {
+      return `Ticket type '${purchases[i].ticketType}' cannot be found.`;
+    }
+    // Return an error message if the ticketEntrant does not exist as a key inside of priceInCents inside of the outer object name of ticketData
+    if (!ticketData[purchases[i].ticketType].priceInCents[purchases[i].entrantType]) {
+      return `Entrant type '${purchases[i].entrantType}' cannot be found.`;
+    }
+
+    // Add the cost of admission to ticketPrice based on the type of ticket purchased and who(entrantType) the ticket is for
+    ticketPrice += ticketData[purchases[i].ticketType].priceInCents[purchases[i].entrantType];
+   
```

Comment:
Another approach you could have taken was to leverage the `calculateTicketPrice` function. If the return is a string then your know it is an error message and return that, if not you would add the price for it ticket to a total .